### PR TITLE
sysfs: adds FS.Stat and companions in platform

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This centralizes filestat logic by making our own `Stat_t` similar to `syscall.Stat_t`. This exposes utilities in the platform package and adds a new function `FS.Stat` which avoids having to use `fs.File` to get the same info. Doing so at the FS abstraction allows us to optimize how it is implemented internally using portable means (e.g. `os.StatFile`) or OS-specific means where necessary, e.g. in windows.

This also ensures `platform.OpenFile` returns syscall.Errno and centralizes error checking with a new `require.EqualErrno` test.

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
I will resolve the glitch tinygo found and backfill a normal test for it:

```
=== RUN   TestStatBadDir
[885](https://github.com/tetratelabs/wazero/actions/runs/4220217670/jobs/7326309383#step:6:886)
--- FAIL: TestStatBadDir (0.00s)
[886](https://github.com/tetratelabs/wazero/actions/runs/4220217670/jobs/7326309383#step:6:887)
    Mkdir error = &T{Op:"stat", Path:"/tmp/not-exist/really-not-exist", Err:0x36}; want PathError for path "/tmp/not-exist/really-not-exist" satisifying IsNotExist
```

**Review by mathetake — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
Err:0x36 is ErrNotDir in wasese

**Comment by codefromthecrypt** [comment]
reworked the PR to ensure `platform.OpenFile` returns syscall.Errno and
centralized error checking with a new `require.EqualErrno` test.

File: internal/testing/require/require.go [hunk 1]
```diff
@@ -126,6 +127,19 @@ func Error(t TestingT, err error, formatWithArgs ...interface{}) {
 	}
 }
 
+// EqualErrno should be used for functions that return syscall.Errno or nil.
+func EqualErrno(t TestingT, expected syscall.Errno, err error, formatWithArgs ...interface{}) {
```
Comment (codefromthecrypt) [inline internal/testing/require/require.go:131]: this is the new assertion, which helps a lot to figure out when different OS choose the same literal errno for different errors!

**Comment by codefromthecrypt** [comment]
sometimes doing stat on a closed file is EIO or EBADF, I'll coerce to the latter, as it is a better errno

**Review by mathetake — APPROVED** [review]

**Pull request merged**